src/plugins/intel_cpu/src/nodes/executors/acl/acl_deconv.cpp

```diff
@@ -13,43 +13,68 @@ using namespace arm_compute;
+            dimensionsMover({&srcVecDims, &weiVecDims, &dstVecDims});
+        }
     }
 
-    TensorInfo srcTensorInfo = TensorInfo(shapeCast(srcDims), 1,
-                                          precisionToAclDataType(srcDescs[0]->getPrecision()), getAclDataLayoutByMemoryDesc(srcDescs[0]));
-    TensorInfo weiTensorInfo = TensorInfo(shapeCast(weiDims), 1,
-                                          precisionToAclDataType(srcDescs[1]->getPrecision()), getAclDataLayoutByMemoryDesc(srcDescs[1]));
-    TensorInfo dstTensorInfo = TensorInfo(shapeCast(dstDims), 1,
-                                          precisionToAclDataType(dstDescs[0]->getPrecision()), getAclDataLayoutByMemoryDesc(dstDescs[0]));
-
-    unsigned int pad_l =
-            (deconvAttrs.paddingL.size() > 1) ? static_cast<unsigned int>(deconvAttrs.paddingL.at(1)) : static_cast<unsigned int>(deconvAttrs.paddingL.at(0));
-    unsigned int pad_r =
-            (deconvAttrs.paddingR.size() > 1) ? static_cast<unsigned int>(deconvAttrs.paddingR.at(1)) : static_cast<unsigned int>(deconvAttrs.paddingR.at(0));
-    unsigned int pad_t = static_cast<unsigned int>(deconvAttrs.paddingL.at(0));
-    unsigned int pad_b = static_cast<unsigned int>(deconvAttrs.paddingR.at(0));
-    unsigned int stride_x = (deconvAttrs.stride.size() > 1) ? deconvAttrs.stride.at(1) : deconvAttrs.stride.at(0);
-    unsigned int stride_y = deconvAttrs.stride.at(0);
-    PadStrideInfo deconv_info(stride_x, stride_y, pad_l, pad_r, pad_t, pad_b, DimensionRoundingType::FLOOR);
+    auto srcLayout = getAclDataLayoutByMemoryDesc(srcDescs[0]);
+    auto weiLayout = getAclDataLayoutByMemoryDesc(srcDescs[1]);
+    auto dstLayout = getAclDataLayoutByMemoryDesc(dstDescs[0]);
+
+    if (srcLayout == arm_compute::DataLayout::NHWC && weiLayout == arm_compute::DataLayout::NCHW) {
+        weiLayout = arm_compute::DataLayout::NHWC;
+    }
+
+    TensorInfo srcTensorInfo = TensorInfo(srcVecDims, 1,
+                                          precisionToAclDataType(srcDescs[0]->getPrecision()), srcLayout);
+    TensorInfo weiTensorInfo = TensorInfo(weiVecDims, 1,
+                                          precisionToAclDataType(srcDescs[1]->getPrecision()), weiLayout);
+    TensorInfo dstTensorInfo = TensorInfo(dstVecDims, 1,
+                                          precisionToAclDataType(dstDescs[0]->getPrecision()), dstLayout);
+    TensorInfo biasTensorInfo;
+    if (deconvAttrs.withBiasesParam) {
+        biasTensorInfo = TensorInfo(biasVecDims, 1,
+                                    precisionToAclDataType(srcDescs[2]->getPrecision()), getAclDataLayoutByMemoryDesc(srcDescs[2]));
+    }
 
     return ACLDeconvTensorInfo{srcTensorInfo, weiTensorInfo, biasTensorInfo, dstTensorInfo, deconv_info};
 }
 
-AclDeconvExecutor::AclDeconvExecutor(const ExecutorContext::CPtr context) : DeconvExecutor(context) {}
+AclDeconvExecutor::AclDeconvExecutor(const ExecutorContext::CPtr context) : DeconvExecutor(context), aclContext(context) {}
```

Comment:
Do we really need `aclContext` given base class also stores pointer on `context` ?

Reply:
@dmitry-gorokhov we don't need it, I removed it